app/components/shared/event-cover-change-field.tsx

```diff
@@ -0,0 +1,58 @@
+import { type OutputFileEntry } from "@uploadcare/blocks"
+import { useState } from "react"
+import { Theme, useTheme } from "remix-themes"
+
+import { defaultLRConfig } from "~/components/libs/uploader-uploadcare"
+import { AvatarUploader } from "~/components/shared/avatar-uploader"
+import { useRootLoaderData } from "~/hooks/use-root-loader-data"
+
+const DEFAULT_IMAGE = "/images/covers/bandungdev-cover-luma-sharing.png"
+
+export function EventCoverChangeField({
+  alt,
+  imageUrl = DEFAULT_IMAGE,
+}: {
+  imageUrl?: string
+  alt: string
+}) {
+  const { ENV } = useRootLoaderData()
+  const [eventCoverFile, setEventCoverFile] = useState<OutputFileEntry | null>(
+    null,
+  )
+
+  const [themeSymbol] = useTheme()
+  const theme = themeSymbol === Theme.DARK ? "dark" : "light"
+
+  if (!ENV.UPLOADCARE_PUBLIC_KEY) return null
+
+  let url = imageUrl
+
+  if (!!eventCoverFile && !!eventCoverFile.cdnUrl) {
+    url = eventCoverFile.cdnUrl
+  }
+
+  return (
+    <div className="relative">
+      <input type="hidden" name="imageUrl" value={url} />
+
+      <img
+        className="aspect-video w-full bg-cover object-cover"
+        alt={alt}
+        src={url}
+        onError={({ currentTarget }) => {
+          currentTarget.onerror = null
+          currentTarget.src = DEFAULT_IMAGE
+        }}
+      />
+      <div className="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2">
+        <AvatarUploader
```

Comment:
@fikrialwan @agus-wesly I think at this point the AvatarUploader need to be renamed into something more general for the form usage.

Reply:
these 2 components basically just the same. I think we can remove one & rename other with more general name

Reply:
oh okay, I've remove one & rename other name to `Uploader` on this commit https://github.com/bandungdevcom/bandungdev.com/pull/42/commits/fa7bb998d020f50c3b8043bc5c62708d297b7240